src/compartment.c

```diff
@@ -57,312 +112,53 @@ comp_from_elf(char *filename, char **entry_points, size_t entry_point_count,
     void *new_comp_base)
 {
     struct Compartment *new_comp = comp_init();
+    new_comp->base = new_comp_base;
+    new_comp->mem_top = new_comp_base;
 
-    new_comp->fd = open(filename, O_RDONLY);
-    if (new_comp->fd == -1)
-    {
-        free(new_comp);
-        errx(1, "Error opening compartment file  %s!\n", filename);
-    }
+    unsigned short libs_to_parse_count = 1;
+    unsigned short libs_parsed_count = 0;
+    char **libs_to_parse = malloc(sizeof(char *));
+    libs_to_parse[0] = filename;
 
-    assert(entry_points);
-    assert(entry_point_count > 0);
-    new_comp->comp_eps
-        = malloc(entry_point_count * sizeof(struct CompEntryPoint));
+    char *libs_folder = getenv(libs_path_env_var);
 
-    // Read elf headers
-    Elf64_Ehdr comp_ehdr;
-    assert(new_comp->fd != -1);
-    do_pread(new_comp->fd, &comp_ehdr, sizeof(Elf64_Ehdr), 0);
-    new_comp->elf_type = comp_ehdr.e_type;
-    if (new_comp->elf_type != ET_DYN)
+    while (libs_parsed_count != libs_to_parse_count)
```

Comment:
I think this is a counting-down-`for` loop in disguise?

Reply:
Do you mean something like `for(; libs_to_parse_count > 0; --libs_to_parse_count)`?

Reply:
Yep.

Reply:
It's clearer because the condition and the decrement end up together (when I looked at the `while` I was looking for a `++` to one variable and only later releaised there's a `--` to another variable!).

Reply:
Sorry, just to check, where is the decrement? I only see increments to both (lines 143 [1] and 148 [2]). 

[1] https://github.com/capablevms/CHERI-ELF-comp/pull/20/files#diff-7887211152112561f3aeba0bbee694021f8d663651ae72fe0164649b2ba2ea32R143
[2] https://github.com/capablevms/CHERI-ELF-comp/pull/20/files#diff-7887211152112561f3aeba0bbee694021f8d663651ae72fe0164649b2ba2ea32R148

Reply:
I personally prefer the `while` version, so I say we leave this as is.